### PR TITLE
Add nocom crash

### PR DESCRIPTION
**Pull request by AidanRB**
## Description
Uses the nocom exploit to cause servers to load lots of chunks at once and lag a lot, or even crash

## (Optional) screenshots / videos
[Not my video but this is it](https://www.youtube.com/watch?v=f0pT6rJnrCA)

### CODE REVIEW

File: src/main/resources/assets/wurst/lang/en_us.json [hunk 1]
```diff
@@ -88,6 +88,7 @@
   "description.wurst.hack.navigator": "A searchable GUI that learns your preferences over time.",
   "description.wurst.hack.nobackground": "Removes the dark background behind inventories.",
   "description.wurst.hack.noclip": "Allows you to freely move through blocks.\nA block (e.g. sand) must fall on your head to activate it.\n\n§c§lWARNING:§r You will take damage while moving through blocks!",
+  "description.wurst.hack.nocomcrash": "Lags servers using the nocom exploit.\nDoes not work on Paper servers.",
```
Comment (Alexander01998) [inline src/main/resources/assets/wurst/lang/en_us.json:91]: If I understand correctly this exploit only works on Spigot servers. If so, the description should mention that.

File: src/main/java/net/wurstclient/hacks/NocomCrashHack.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import net.minecraft.util.math.Vec3d;
+import net.wurstclient.Category;
+import net.wurstclient.SearchTags;
+import net.wurstclient.hack.Hack;
+import net.wurstclient.settings.SliderSetting;
+import net.wurstclient.settings.SliderSetting.ValueDisplay;
+import net.wurstclient.util.ChatUtils;
+
+@SearchTags({ "server crasher", "nocom crash",
+		"1.18.1" })
```
Comment (Alexander01998) [inline src/main/java/net/wurstclient/hacks/NocomCrashHack.java:28]: Is this really exclusive to 1.18.1? I thought the NoCom exploit was used on the 2b2t server running 1.12.2?

**Review by Alexander01998 — CHANGES_REQUESTED** [review]
Please clarify exactly what type of servers this hack is compatible with. I need this info for documentation and to set up test servers.

**Comment by AidanRB** [comment]
From my testing, this works on vanilla, spigot, and fabric servers. Paper has it patched. I don't know what versions it works on, except that it works in 1.18.1. I assume it works for many older versions as well. I'll try to stand up some test servers and test it out this weekend.

**Comment by AidanRB** [comment]
Added clearer documentation on the tested working server types

**Review by Alexander01998 — APPROVED** [review]

**Comment by Saturn5Vfive** [comment]
This hack was originall made by my friend 0x150 (his github got locked) You should at least add credit to him somewhere instead of this guy who did almost 0 work pasting the code into wurst.

**Comment by 0x3C50** [comment]
Yeah I was about to say, this is my content, and I do not want this here. Can we speak in private about this? 0x150 the 2nd#0150

**Comment by Alexander01998** [comment]
@0x3C50

1. @AidanRB has signed a legally binding [agreement](https://cla-assistant.io/Wurst-Imperium/Wurst7?pullRequest=560) to confirm that this hack is either their original creation, or that they have all the necessary rights to submit this hack and have pointed out any original sources, licenses, or restrictions that they are aware of.
2. Copyright only applies to actual work, not to abstract ideas. You cannot copyright the idea of a NocomCrash hack, just like nobody can copyright the idea of a Killaura, or a Minecraft Hacked Client. Copyright applies when someone literally copies another person's work.
3. Looking at [your implementation](https://github.com/0x3C50/Coffee/blob/master/src/main/java/me/x150/sipprivate/feature/module/impl/exploit/NoComCrash.java), it looks substantially different from what was [submitted here](https://github.com/Wurst-Imperium/Wurst7/pull/560/files). Neither version seems to be a direct copy of the other. The only parts that are the same are the ones that have to be the same in order for the hack to function.
4. For example, @AidanRB's version does not use an ItemStack or CreativeInventoryActionC2SPacket, while your version does not use a separate Thread. I am confident that @AidanRB has put their own work into developing and testing this hack, and there is clear evidence to support this.

**Comment by AidanRB** [comment]
@0x3C50 Hey, trying to add you but you have friend requests off. My intention was not to steal your code, I was under the impression you were okay with it. If I can add you we can talk.

**Comment by 0x3C50** [comment]
> @0x3C50
> 
>     1. @AidanRB has signed a legally binding [agreement](https://cla-assistant.io/Wurst-Imperium/Wurst7?pullRequest=560) to confirm that this hack is either their original creation, or that they have all the necessary rights to submit this hack and have pointed out any original sources, licenses, or restrictions that they are aware of.
> 
>     2. Copyright only applies to actual work, not to abstract ideas. You cannot copyright the idea of a NocomCrash hack, just like nobody can copyright the idea of a Killaura, or the idea of a Minecraft Hacked Client. Copyright applies when someone literally copies another person's work.
> 
>     3. Looking at [your implementation](https://github.com/0x3C50/Coffee/blob/master/src/main/java/me/x150/sipprivate/feature/module/impl/exploit/NoComCrash.java), it looks substantially different from what was [submitted here](https://github.com/Wurst-Imperium/Wurst7/pull/560/files). Neither version seems to be a direct copy of the other. The only parts that are the same are the ones that have to be the same in order for the hack to function.
> 
>     4. For example, @AidanRB's version does not use an ItemStack or CreativeInventoryActionC2SPacket, while your version does not use a separate Thread. I am confident that @AidanRB has put their own work into developing and testing this hack, and there is clear evidence to support this.

My mod has 2 different modes, both of them work essentially the same. The same concept was applied here, even copying the random method name and the variable it gets saved under:

This implementation
image
image


The original
image
image

**Comment by 0x3C50** [comment]
> @0x3C50 Hey, trying to add you but you have friend requests off. My intention was not to steal your code, I was under the impression you were okay with it. If I can add you we can talk. crisp#5610

Also sorry for taking so long, sent the friend request

**Comment by 0x3C50** [comment]
@Alexander01998 can we go into a vc and discuss this please? This is becoming very interesting very fast.

**Comment by Alexander01998** [comment]
@0x3C50 @Saturn5Vfive No, I will not go into voice chat with you. I have seen enough of your disrespectful and immature behavior in the comments above. Either you figure this out with @AidanRB or we're done here.

**Pull request merged**